spec/std/set_spec.cr

```diff
@@ -74,6 +74,24 @@ describe "Set" do
     x.to_a.should eq([1])
   end
 
+  it "checks intersect" do
+    set = Set{3, 4, 5}
+    empty_set = Set(Int32).new
+
+    set.intersect?(set).should be_true
+    set.intersect?(Set{2, 4}).should be_true
+    set.intersect?(Set{5, 6, 7}).should be_true
+    set.intersect?(Set{1, 2, 6, 8, 4}).should be_true
+
+    set.intersect?(empty_set).should be_false
+    set.intersect?(Set{0, 2}).should be_false
+    set.intersect?(Set{0, 2, 6}).should be_false
+    set.intersect?(Set{0, 2, 6, 8, 10}).should be_false
```

Comment:
Imo the different checks/assertions should be split up into multiple test cases - makes it easier to identify what broke if something broke :)